File: test-suite/golden-tests/collect-formatters/impl.py
def make_ints(i):
    return [i * 10 + 3, i * 10 + 1, i * 10 + 2]

def sort_ints(xs):
    return sorted(xs)

def tag_ints(xs):
    return xs + [len(xs)]

def tag_ints_off(off, xs):
    return [off] + xs

def render_ints(xs):
    return " ".join(str(x) for x in xs) + " \n"

def render_ints_off(off, xs):
    return "[" + str(off) + "]" + "".join(" " + str(x) for x in xs) + "\n"

def make_strs(i):
    return ["b" + str(i), "a" + str(i)]

def sort_strs(xs):
    return sorted(xs)

def join_strs(xs):
    return ";".join(xs) + ";\n"

File: test-suite/golden-tests/collect-formatters/impl.R
make_ints <- function(i) {
  c(i * 10 + 3, i * 10 + 1, i * 10 + 2)
}

sort_ints <- function(xs) {
  sort(xs)
}

tag_ints <- function(xs) {
  c(xs, length(xs))
}

tag_ints_off <- function(off, xs) {
  c(off, xs)
}

render_ints <- function(xs) {
  paste0(paste(xs, collapse = " "), " \n")
}

render_ints_off <- function(off, xs) {
  paste0("[", off, "]", paste0(" ", xs, collapse = ""), "\n")
}

make_strs <- function(i) {
  c(paste0("b", i), paste0("a", i))
}

sort_strs <- function(xs) {
  sort(xs)
}

join_strs <- function(xs) {
  paste0(paste(xs, collapse = ";"), ";\n")
}
